client/packages/invoices/src/OutboundShipment/DetailView/AddFromScannerButton.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect } from 'react';
+import {
+  useTranslation,
+  useToggle,
+  InvoiceNodeStatus,
+  useBarcodeScannerContext,
+  CircularProgress,
+  ScanIcon,
+  ScanResult,
+  ButtonWithIcon,
+  useNotification,
+} from '@openmsupply-client/common';
+import {
+  ItemRowFragment,
+  MasterListSearchModal,
+} from '@openmsupply-client/system';
+import { useOutbound } from '../api';
+
+export const AddFromScannerButtonComponent = ({
+  onAddItem,
+}: {
+  onAddItem: (item?: ItemRowFragment) => void;
+}) => {
+  const t = useTranslation('distribution');
+  const { status } = useOutbound.document.fields(['status']);
+  const isDisabled = status !== InvoiceNodeStatus.New;
+  const { addFromMasterList } = useOutbound.utils.addFromMasterList();
+  const { mutateAsync: getBarcodes } = useOutbound.utils.barcodes();
+  const { otherPartyId } = useOutbound.document.fields(['otherPartyId']);
+  const modalController = useToggle();
+  const filterByName = { existsForNameId: { equalTo: otherPartyId } };
+  const { hasBarcodeScanner, isScanning, startScanning, stopScan } =
+    useBarcodeScannerContext();
+  const { warning } = useNotification();
+
+  const handleScanResult = async (result: ScanResult) => {
+    if (!!result.content) {
+      const { content, gtin } = result;
+      const barcode = gtin ?? content;
+      const barcodes = await getBarcodes(barcode);
+
+      if (barcodes.totalCount === 0) {
+        warning(t('error.no-matching-item'))();
+        // TODO: save barcode after selection
```

Comment:
Where would it save to? Invoice Line?

Reply:
Just saving back to the barcode table, against the selected item_id. This allows lookup for next time

Reply:
Looks like this might be done in the next PR ?

Reply:
yeah, is in the next PR